cfme/physical/physical_chassis.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+"""A model of an Infrastructure PhysicalChassis in CFME."""
+import attr
+
+from lxml.html import document_fromstring
+from navmazing import NavigateToAttribute
+from widgetastic_patternfly import Dropdown, Accordion
+from widgetastic.widget import Text, View
+from wrapanapi.entities import PhysicalContainer
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.common import PolicyProfileAssignable, Taggable
+from cfme.exceptions import (
+    ItemNotFound,
+    StatsDoNotMatch,
+    HostStatsNotContains,
+    ProviderHasNoProperty
+)
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.log import logger
+from cfme.utils.pretty import Pretty
+from cfme.utils.providers import get_crud_by_name
+from cfme.utils.update import Updateable
+from cfme.utils.varmeth import variable
+from widgetastic_manageiq import (
+    BaseEntitiesView,
+    JSBaseEntity,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    ParametrizedSummaryTable,
+    ManageIQTree
+)
+
+
+@attr.s
+class PhysicalChassis(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Chassis in cfme.
+
+    Args:
+        name: Name of the physical chassis.
+        custom_ident: The custom identifiter.
+
+    """
+    pretty_attrs = ['name', 'custom_ident']
+
+    name = attr.ib()
+    ems_ref = attr.ib(default=None)
+    provider = attr.ib(default=None)
```

Comment:
This is going to seem like a nitpick, and I was hoping to merge this today, but `provider` really needs to come before `ems_ref` for the instance attributes to maintain consistency when instantiating the entities.

Reply:
👍